torchrl/envs/libs/pettingzoo.py

```diff
@@ -608,8 +606,50 @@ def _step(
                         " you need to set use_action_mask=True to allow this."
                     )
 
+        # set done values
+        done, terminated, truncated = self._aggregate_done(
+            tensordict_out, use_any=self.parallel
+        )
+
+        tensordict_out.set("done", done)
+        tensordict_out.set("terminated", terminated)
+        tensordict_out.set("truncated", truncated)
         return tensordict_out
 
+    def _aggregate_done(self, tensordict_out, use_any):
+        done = False
+        truncated = False
+        terminated = False
+        for key in self.full_done_spec.keys(True, True):
+            if isinstance(key, tuple):
+                if use_any:
+                    if key[-1] == "done":
+                        done = done | tensordict_out.get(key).any()
+                    if key[-1] == "terminated":
+                        terminated = terminated | tensordict_out.get(key).any()
+                    if key[-1] == "truncated":
+                        truncated = truncated | tensordict_out.get(key).any()
+                    if done:
+                        break
+                else:
+                    if key[-1] == "done":
+                        done = done & tensordict_out.get(key).all()
+                    if key[-1] == "terminated":
+                        terminated = terminated & tensordict_out.get(key).all()
+                    if key[-1] == "truncated":
+                        truncated = truncated & tensordict_out.get(key).all()
+                    if not done:
+                        break
+        else:
+            raise RuntimeError(
```

Comment:
this else branch is currrently reached every time the for loop finishes

I would remove it

Reply:
I think the problem is that the loop is broken, but this should stay. If there's no done signal something defo went wrong